File: Backend/Elevate.Data/Models/Habit/HabitModel.cs
using Elevate.Models;

namespace Elevate.Data.Models.Habit
{
    public class HabitModel
    {
        public required Guid Id { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public required string Title { get; set; }
        public string? Description { get; set; }
        public required FrequencyEnum FrequencyType { get; set; }
        public int? CustomFrequency { get; set; }
        public required string Color { get; set; }
        public required bool IsPositive { get; set; }
        public int Streak { get; set; } = 0;
        public DateTime StreakStart { get; set; } = DateTime.UtcNow;
        public required bool Deleted { get; set; }
    }
}

File: web-frontend/src/app/models/habit.model.ts
export interface Habit {
  id: string;
  userId: string;
  createdAt: Date;
  title: string;
  description?: string;
  frequencyType: FrequencyEnum;
  customFrequency?: number;
  color: string;
  isPositive: boolean;
  streak: number;
  streakStart?: Date;
  deleted: boolean;
  streakProgression: string; // Format: "x/y" where x is completed and y is total
}

export interface NegativeHabit {
  id: string;
  userId: string;
  updatedAt: string | Date;
  title: string;
  description?: string;
  color: string;
}

export enum FrequencyEnum {
  Daily = 'Daily',
  Weekly = 'Weekly',
  Monthly = 'Monthly',
  Custom = 'Custom',
}
